utilities/PerformanceTesting/GetAverages.py

```diff
@@ -0,0 +1,146 @@
+
+    :param jsons: A list of jsons containing metrics
+    :param metric_group: The metric group that the metric belongs to
+    :param metric: The metric that is being averaged
+
+    :return: Returns a tuple. The first element is the average or 0
+    if it is not possible to calculate the average.
+    The second element is a boolean specifying if the average was
+    successfully computed.
+    """
+    total = 0
+    count = 0
+
+    for j in jsons:
+        if metric not in j[metric_group]:
+            err_msg = "Missing metric: " + metric
+            raise KeyError(err_msg)
+
+        try:
+            value = j[metric_group][metric]['value']
+            total += float(value)
+            count += 1.0
+        except (ValueError, TypeError):
+            continue
+
+    if count > 0:
+        return total / float(count), True
+    else:
+        return 0, False
+
+
+def calcAverage(file_prefix, num_files):
+    """
+    Computes the average results over multiple runs
+
+    :param file_prefix: The first part of the file path
+    that all files to be averaged have in common
+    :param num_files: The number of files to be averaged
+
+    :return: A json with the average of the results of the different runs
+    """
+
+    files = getFilePaths(file_prefix, num_files)
+
+    jsons = getJsons(files)
+    ave_json = jsons[0]
+
+    for metric_group in jsons[0]:
+        for metric in jsons[0][metric_group]:
+            ave, valid = findAverageForMetric(jsons, metric_group, metric)
+            if valid:
+                ave_json[metric_group][metric]['value'] = str(ave)
+
+    return ave_json
+
+
+if __name__ == '__main__':
+    description = 'Calculates the average of metrics '
+    description += 'from multiple runs of StackRox'
+    parser = argparse.ArgumentParser(
+                                    description=description,
+                                    formatter_class=RawTextHelpFormatter
+                                    )
+
+    file_prefix_help = 'The json file paths to be averaged over '
+    file_prefix_help += 'not including index and extension.\n'
+    file_prefix_help += 'The following is an example of what an \n'
+    file_prefix_help += 'input file might look like\n'
+    file_prefix_help += '{\n'
+    file_prefix_help += '\tmetric_group_1: {\n'
+    file_prefix_help += '\t\t"metric_1a": {\n'
+    file_prefix_help += '\t\t\t"value": 123\n'
+    file_prefix_help += '\t\t},\n'
+    file_prefix_help += '\t\t"metric_1b": {\n'
+    file_prefix_help += '\t\t\t"value": 567\n'
+    file_prefix_help += '\t\t}\n'
+    file_prefix_help += '\t},\n'
+    file_prefix_help += '\tmetric_group_2: {\n'
+    file_prefix_help += '\t\t"metric_2a": {\n'
+    file_prefix_help += '\t\t\t"value": 234\n'
+    file_prefix_help += '\t\t},\n'
+    file_prefix_help += '\t\t"metric_2b": {\n'
+    file_prefix_help += '\t\t\t"value": 789\n'
+    file_prefix_help += '\t\t}\n'
+    file_prefix_help += '\t}\n'
+    file_prefix_help += '}\n'
```

Comment:
We can use a multi-line string here for clarity:

```suggestion
file_prefix_help="""
The json file paths to be averaged over, not including
index and extension.

The following is an example of what an input file might look like:
{
    "metric_group_1": {
        "metric_1a": {
            "value": 123
        },
        "metric_1b": {
            "value": 567
        }
    }
    "metric_group_2": {
        "metric_2a": {
            "value": 234
        },
        "metric_2b": {
            "value": 789
        }
    }
}
"""
```

Reply:
Done